src/FormNetworks/bilinearformnetwork.jl

```diff
@@ -0,0 +1,61 @@
+default_bra_vertex_map(v) = (v, "bra")
+default_ket_vertex_map(v) = (v, "ket")
+default_operator_vertex_map(v) = (v, "operator")
+default_operator_constructor(s::IndsNetwork) = delta_network(s)
+
+struct BilinearFormNetwork{
+  V,TensorNetwork<:AbstractITensorNetwork{V},BraMap,KetMap,OperatorMap
+} <: AbstractFormNetwork{V}
+  tensornetwork::TensorNetwork
+  operator_vertex_map::OperatorMap
+  bra_vertex_map::BraMap
+  ket_vertex_map::KetMap
+end
+
+function BilinearFormNetwork(
+  operator::AbstractITensorNetwork,
+  bra::AbstractITensorNetwork,
+  ket::AbstractITensorNetwork;
+  operator_vertex_map=default_operator_vertex_map,
+  bra_vertex_map=default_bra_vertex_map,
+  ket_vertex_map=default_ket_vertex_map,
+)
+
+  # TODO: Reminder to fix `rename_vertices(::AbstractITensorNetwork)`.
+  bra_renamed = rename_vertices_itn(bra, bra_vertex_map)
+  ket_renamed = rename_vertices_itn(ket, ket_vertex_map)
+  operator_renamed = rename_vertices_itn(operator, operator_vertex_map)
+
+  # TODO: Reminder to fix `union` so that `union(bra_renamed, operator_renamed, ket_renamed)` works.
+  tn = union(union(bra_renamed, operator_renamed), ket_renamed)
```

Comment:
It doesn't support a general function map of the names, just a vertex name that you will append to the current vertices to define the new vertex names.

I think it could be nice to use that. It would basically just require changing the vertex maps to just strings, say:
```julia
default_bra_vertex_suffix(v) = "bra"
default_ket_vertex_suffix(v) = "ket"
default_operator_vertex_suffix(v) = "operator"
```
then store the suffixes in `BilinearForm` instead of the maps.

Reply:
Unfortunately `disjoint_union` seems to be broken for more than two inputs, i.e.:
```julia
⊔("x" => g1, "y" => g2, "z" => g3)
```
throws an error inside `union`, but that is easily fixed with:
```julia
Base.union(graph1::AbstractNamedGraph, graph2::AbstractNamedGraph, graph3::AbstractNamedGraph, graph_rest::AbstractNamedGraph...) = union(union(graph1, graph2), graph3, graph_rest...)
```

Reply:
In the future we could also support general maps as part of `disjoint_union`, i.e. `⊔((v -> (v, "x")) => g1, (v -> (v, "y")) => g2, (v -> (v, "z")) => g3)` and we could then go back to storing vertex maps instead of vertex suffixes, but suffixes seem sufficient for now.

Reply:
Also to preserve some of the current functionality, we could define `bra_vertex_map(f::AbstractForm) = (v -> (v, bra_vertex_suffix(f))`, which will probably be nice to have anyway.

Reply:
Okay nice yeah this looks neater. Funnily the `disjoint_union(suffix => tn::ITensorNetwork)` seems to automatically work. So it is renaming it correctly. But `rename_vertices(tn::ITensorNetwork, map)` doesn't work because `rename_vertices` doesn't map

Reply:
Actually `rename_vertices` does wrap because it is defined in `AbstractITensorNetwork.jl` which is why `disjoint_union` automatically works. Hence, I don't know why `rename_vertices_itn` is defined at all. I will delete it.